build.sbt

```diff
@@ -26,6 +26,12 @@ def inferJavaHome() = {
   Some(actualHome)
 }
 
+lazy val scala3ScalacOption = Def.setting (
```

Comment:
there is a much cleaner way to do this:
- since the decision must be taken at the project level and not at the build level, remove https://github.com/scalacenter/scalafix/blob/59ed8c053a79cadfe3ef2f83a10561df7170a192/build.sbt#L8
- add the opposite of this (i.e. adding `-P:semanticdb:synthetics:on` if not scala3) in https://github.com/scalacenter/scalafix/blob/59ed8c053a79cadfe3ef2f83a10561df7170a192/project/ScalafixBuild.scala#L180
- you can now get rid of all the customizations!

Reply:
having the reverse logic of adding the option only when needed is more readable, i like it!